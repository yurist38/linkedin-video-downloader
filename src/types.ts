export enum Actions {
  AddButtons = 'addButtons',
  Download = 'download',
}

export interface Message {
  action: Actions;
  url?: string;
}

export interface Options {
  buttonPosition: {
    horizontal: 'left' | 'right';
    vertical: 'top' | 'bottom';
  };
}

export enum CommonNames {
  ButtonClassName = 'linkedin-video-downloader-btn',
  OptionsStorageKey = 'linkedin-video-downloader-options',
}
