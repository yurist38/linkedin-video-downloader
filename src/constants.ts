import { Options } from './types';

export const defaultOptions: Options = {
  buttonPosition: {
    horizontal: 'left',
    vertical: 'top',
  }
};
