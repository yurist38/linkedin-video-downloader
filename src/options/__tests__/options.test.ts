import { mount, VueWrapper, config } from '@vue/test-utils';
import Options from '../options.vue';
import { CommonNames } from '../../types';
import { defaultOptions } from '../../constants';

config.global.components = {
  'it-button': {
    template: '<div>it-button</div>',
  },
  'it-radio': {
    template: '<div>it-radio</div>',
  },
  'it-divider': {
    template: '<div>it-divider</div>',
  },
  'it-input': {
    template: '<div>it-input</div>',
  }
};

describe('options/options', () => {
  let wrapper: VueWrapper;

  beforeAll(() => {
    chrome.storage.local.get = jest.fn();
    chrome.storage.local.set = jest.fn();
  });

  it('Should render correctly', () => {
    wrapper = mount(Options);

    expect(wrapper.element).toMatchSnapshot();

    expect(chrome.storage.local.get).toHaveBeenCalledTimes(1);
    expect((chrome.storage.local.get as jest.Mock).mock.calls[0][0])
      .toEqual([CommonNames.OptionsStorageKey]);
    expect(wrapper.vm.$data.options).toEqual(defaultOptions);
    expect(wrapper.vm.$data.areOptionsChanged).toEqual(false);

    (chrome.storage.local.get as jest.Mock).mock.calls[0][1]();
    expect(wrapper.vm.$data.options).toEqual(defaultOptions);

    (chrome.storage.local.get as jest.Mock).mock.calls[0][1]({});
    expect(wrapper.vm.$data.options).toEqual(defaultOptions);

    (chrome.storage.local.get as jest.Mock).mock.calls[0][1]({
      [CommonNames.OptionsStorageKey]: {
        key: 'value'
      },
    });
    expect(wrapper.vm.$data.options.key).toEqual('value');

    wrapper.vm.saveChanges();
    (chrome.storage.local.set as jest.Mock).mock.calls[0][1]();
    expect(chrome.storage.local.set).toHaveBeenCalledTimes(1);
    expect(wrapper.vm.$data.areOptionsChanged).toEqual(false);
  });

  it('Should set form to invalid state when the filename is empty', () => {
    process.env.APPZI_TOKEN = 'test-value';
    process.env.APPZI_ID = 'test-value';

    wrapper = mount(Options);

    wrapper.vm.onChangeOption();
    expect(wrapper.vm.$data.areOptionsChanged).toEqual(true);
    expect(wrapper.vm.$data.areOptionsValid).toEqual(true);
  });
});
