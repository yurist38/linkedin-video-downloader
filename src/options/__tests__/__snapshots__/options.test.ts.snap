// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options/options Should render correctly 1`] = `
<section
  class="content"
>
  <div
    class="centered"
  >
    <img />
    <h1
      class="title"
    >
      LinkedIn Video Downloader
    </h1>
  </div>
  <div
    class="option"
  >
    <div
      class="option-title"
    >
      Button's vertical position:
    </div>
    <div>
      <div
        label="Top"
        modelvalue="top"
        name="button-position-vertical"
        value="top"
      >
        it-radio
      </div>
      <div
        label="Bottom"
        modelvalue="top"
        name="button-position-vertical"
        value="bottom"
      >
        it-radio
      </div>
    </div>
  </div>
  <div
    class="option"
  >
    <div
      class="option-title"
    >
      Button's horizontal position:
    </div>
    <div>
      <div
        label="Left"
        modelvalue="left"
        name="button-position-horizontal"
        value="left"
      >
        it-radio
      </div>
      <div
        label="Right"
        modelvalue="left"
        name="button-position-horizontal"
        value="right"
      >
        it-radio
      </div>
    </div>
  </div>
  <div
    class="option"
  >
    <div
      class="option-title"
    >
      Filename:
    </div>
    <div
      modelvalue="linkedin-video"
      name="filename-input"
    >
      it-input
    </div>
  </div>
  <div
    class="btn center"
    disabled="true"
    outlined=""
    type="primary"
  >
    it-button
  </div>
</section>
`;
